File: Frontend/src/app/viewModels/requests/ResetPasswordVM.ts
import { toJSONString } from "../interfaces/toJSONString";
import { ChangePasswordVM } from "./ChangePasswordVM";

/**
 * View Model for a reset password request
 */
export class ResetPasswordVM extends ChangePasswordVM implements toJSONString {

    /**
     * Construct's an instance of {@link ResetPasswordVM}
     * 
     * @param password new password
     * @param email user's email
     */
    constructor(
        public password: string,
        public email: string
    ) {
        super(password)
    }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify(this)
    }

}

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Request/ResetPasswordVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View model class for the reset password last step request. 
    /// This class represents the last step where the user sends the new password.
    /// This class extends ChangePasswordVM once the only difference is that in this one 
    /// we also need to have the user's email.
    /// </summary>
    public class ResetPasswordVM : ChangePasswordVM
    {
        public string Email { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="email">User's email</param>
        /// <param name="password">User's password</param>
        public ResetPasswordVM(string email, string password) : base(password)
        {
            this.Email = email;
        }
    }
}